### ISSUE TITLE
Map and Set forEach throw errors

### ISSUE THREAD

**Issue by hpx7** [sindresorhus/on-change#73]
```
TypeError: Method Map.prototype.forEach called on incompatible receiver [object Object]
    at Proxy.forEach (<anonymous>)
```

Similar to https://github.com/sindresorhus/on-change/issues/67

**Comment by hpx7** [comment]
@DarrenPaulWright curious if you have thoughts on how to fix this? I guess we would need to add `forEach` [here](https://github.com/sindresorhus/on-change/blob/6a14ae602d19eae92a6fec0d9290e9089c4a7337/lib/smart-clone.js#L89-L93) but not sure if anything would be required as well.

**Comment by DarrenPaulWright** [comment]
This is fixed in 3.0.0
